eth/client.go

```diff
@@ -883,84 +888,31 @@ func (c *client) ContractAddresses() map[string]ethcommon.Address {
 }
 
 func (c *client) CheckTx(tx *types.Transaction) error {
-	ctx, cancel := context.WithTimeout(context.Background(), c.txTimeout)
-	defer cancel()
-
-	receipt, err := bind.WaitMined(ctx, c.backend, tx)
-	if err != nil {
-		return err
+	txs := make(chan *types.Receipt, 10)
+	b, ok := c.backend.(*backend)
```

Comment:
Could we avoid this type assertion if the TransactionManager was just a field on `client`? Feels a bit weird to do a type assertion just so we can fetch the TransactionManager field on the underlying backend type.

Reply:
The contract bindings use `Backend` to send transactions which uses `TransactionManager` under the hood. 

The `Backend` does not have access to the `client`.